main.go

```diff
@@ -66,53 +72,117 @@ const (
+	var err error
+	var tlsProfile *ocpconfigv1.TLSSecurityProfile
 
-	minTlsVersion, err := tlsOptions.MinTLSVersionId()
-	if err != nil {
-		return err
+	var sspList ssp.SSPList
+	if err := apiClient.List(ctx, &sspList, &client.ListOptions{}); err != nil {
+		return nil, err
 	}
 
-	tlsConfig := tls.Config{
-		CipherSuites: tlsOptions.CipherIDs(&setupLog),
-		MinVersion:   minTlsVersion,
+	if len(sspList.Items) != 0 {
+		tlsProfile = sspList.Items[0].Spec.TLSSecurityProfile
 	}
 
-	server := http.Server{
-		Addr:      metricsAddr,
-		Handler:   mux,
-		TLSConfig: &tlsConfig,
+	if tlsProfile == nil {
+		cfg.MinVersion = crypto.DefaultTLSVersion()
+		cfg.CipherSuites = nil
+		return cfg, nil
 	}
 
+	if tlsProfile.Type == ocpconfigv1.TLSProfileCustomType {
+		cfg.CipherSuites = common.CipherIDs(tlsProfile.Custom.Ciphers, &setupLog)
+		cfg.MinVersion, err = crypto.TLSVersion(string(tlsProfile.Custom.MinTLSVersion))
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cipherNames, minTypedTLSVersion := ocpconfigv1.TLSProfiles[tlsProfile.Type].Ciphers, ocpconfigv1.TLSProfiles[tlsProfile.Type].MinTLSVersion
+	cfg.CipherSuites = common.CipherIDs(cipherNames, &setupLog)
+	cfg.MinVersion, err = crypto.TLSVersion(string(minTypedTLSVersion))
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+type prometheusServer struct {
+	server   *http.Server
+	client   client.Client
+	certPath string
+	keyPath  string
+}
+
+// InjectClient implements the interface that is used to inject
+// the Client dependency initialized by the ControllerManager.
+func (s *prometheusServer) InjectClient(client client.Client) error {
+	s.client = client
+	return nil
+}
+
+// NeedLeaderElection implements the LeaderElectionRunnable interface, which indicates
+// the prometheus server doesn't need leader election.
+func (s *prometheusServer) NeedLeaderElection() bool {
+	return false
+}
+
+func (s *prometheusServer) Start(ctx context.Context) error {
 	go func() {
-		err := server.ListenAndServeTLS(path.Join(sdkTLSDir, sdkTLSCrt), path.Join(sdkTLSDir, sdkTLSKey))
+		err := s.server.ListenAndServeTLS(s.certPath, s.keyPath)
 		if err != nil {
 			setupLog.Error(err, "Failed to start Prometheus metrics endpoint server")
 		}
 	}()
 	return nil
 }
 
-func getWebhookServer(sspTLSOptions common.SSPTLSOptions) *webhook.Server {
-	// If TLSSecurityProfile is empty, we want to return nil so that the default
-	// webhook server configuration is used.
-	if sspTLSOptions.IsEmpty() {
-		return nil
+func (s *prometheusServer) getPrometheusTLSConfig(ctx context.Context, certWatcher *certwatcher.CertWatcher) *tls.Config {
+	return &tls.Config{
+		GetConfigForClient: func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+			cfg := &tls.Config{}
+			cfg.GetCertificate = certWatcher.GetCertificate
+			return getConfigForClientCallback(cfg, s.client, ctx)
+		},
+	}
+}
+
+func getPrometheusServer(metricsAddr string, ctx context.Context) (*prometheusServer, error) {
```

Comment:
Can you rename this function to `newPrometheusServer()`? Because it is a constructor.